pkg/detectors/aws/aws.go

```diff
@@ -140,10 +142,18 @@ func (s Scanner) FromData(ctx context.Context, verify bool, data []byte) (result
 
 				res, err := client.Do(req)
 				if err == nil {
-					// Response body is unused so close it immediately.
-					res.Body.Close()
+
 					if res.StatusCode >= 200 && res.StatusCode < 300 {
-						s1.Verified = true
+						identityInfo := identityRes{}
+						err := json.NewDecoder(res.Body).Decode(&identityInfo)
+						if err == nil {
+							s1.Verified = true
+							s1.ExtraData = make(map[string]string)
+							s1.ExtraData["account"] = identityInfo.GetCallerIdentityResponse.GetCallerIdentityResult.Account
+							s1.ExtraData["user_id"] = identityInfo.GetCallerIdentityResponse.GetCallerIdentityResult.UserID
+							s1.ExtraData["arn"] = identityInfo.GetCallerIdentityResponse.GetCallerIdentityResult.Arn
```

Comment:
**Nit (optional):** Grouping this in one assignment is slightly more favorable

```go
s1.ExtraData = map[string]string{
    "account": identityInfo...
    "user_id": identityInfo...
    "arn": identityInfo...
}
```

Reply:
agreed!